fuel/app/classes/materia/perm/manager.php

```diff
@@ -684,16 +684,19 @@ static public function get_all_users_explicit_perms($object_id, $object_type)
 	 * @param unknown NEEDS DOCUMENTATION
 	 * @return bool, true if removed object permissions, false if current user does not have owner premissions
 	 */
-	static public function remove_all_permissions($object_id, $object_type)
+	static public function remove_all_permissions($object_id, $object_type, $require_ownership = true)
 	{
 		// make sure the current user has rights to this item
 		// NOTE: might want to fix stuff to re-enable this
 		// there was a bug upon copying a game when the permissions to the copy wer being re-set
 		$current_user_id = (int)\Model_User::find_current_id();
 
-		if ( ! self::get_user_object_perms($object_id, $object_type, $current_user_id)) return false;
+		if ($require_ownership && ! self::get_user_object_perms($object_id, $object_type, $current_user_id)) return false;
 
-		\Event::trigger('delete_widget_event', ['user_id' => $current_user_id, 'object_id' => $object_id, 'object_type' => $object_type]);
+		if ($require_ownership)
+		{
+			\Event::trigger('delete_widget_event', ['user_id' => $current_user_id, 'object_id' => $object_id, 'object_type' => $object_type]);
```

Comment:
In the current code, `Perm_Manager::remove_all_permissions($this->id, Perm::INSTANCE);` short-circuits early, specifically during this check: 
```
if ( ! self::get_user_object_perms($object_id, $object_type, $current_user_id)) return false;
```

It never gets to the `delete_widget_event` event trigger. But if it did, the "widget deleted" notification would be displayed for any user who also had permissions to a widget when it was copied, because of how we're stripping all permissions from the duplicate before re-applying them. The easiest solution was to make that notification contingent on whether the current user actually owned the widget being deleted, so it wouldn't fire during the widget copy process.